pkg/boot/netboot/netboot.go

```diff
@@ -38,32 +38,32 @@ import (
 //
 // TODO: detect straight up multiboot and bzImage Linux kernel files rather
 // than just configuration scripts.
-func BootImage(s curl.Schemes, lease dhclient.Lease) (*boot.LinuxImage, error) {
+func BootImage(l ulog.Logger, s curl.Schemes, lease dhclient.Lease) (*boot.LinuxImage, error) {
 	uri, err := lease.Boot()
 	if err != nil {
 		return nil, err
 	}
-	log.Printf("Boot URI: %s", uri)
+	l.Printf("Boot URI: %s", uri)
 
 	// IP only makes sense for v4 anyway, because the PXE probing of files
 	// uses a MAC address and an IPv4 address to look at files.
 	var ip net.IP
 	if p4, ok := lease.(*dhclient.Packet4); ok {
 		ip = p4.Lease().IP
 	}
-	return getBootImage(s, uri, lease.Link().Attrs().HardwareAddr, ip)
+	return getBootImage(l, s, uri, lease.Link().Attrs().HardwareAddr, ip)
 }
 
 // getBootImage attempts to parse the file at uri as an ipxe config and returns
 // the ipxe boot image. Otherwise falls back to pxe and uses the uri directory,
 // ip, and mac address to search for pxe configs.
-func getBootImage(schemes curl.Schemes, uri *url.URL, mac net.HardwareAddr, ip net.IP) (*boot.LinuxImage, error) {
+func getBootImage(l ulog.Logger, schemes curl.Schemes, uri *url.URL, mac net.HardwareAddr, ip net.IP) (*boot.LinuxImage, error) {
 	// Attempt to read the given boot path as an ipxe config file.
-	ipc, err := ipxe.ParseConfigWithSchemes(uri, schemes)
+	ipc, err := ipxe.ParseConfigWithSchemes(l, uri, schemes)
```

Comment:
todo: would be nice if we could, as standard pxe seems to do on the atonic pi, see if the file is of a bootable type and  try to boot it. That's handy. We could even get lazy and see if a kexec would work on it ... the kernel will tell us if not.

Reply:
yeah, that's actually a necessary step for us to consider merging this code path with fbnetboot's -- that's how they boot